hack/verify-config.sh

```diff
@@ -39,3 +40,6 @@ if [ ! -z "$DIFF" ]; then
     echo "FAILED: config is not correct, please run 'hack/update-config.sh'"
     exit 1
 fi
+
+# Run checkconfig with strict warnings.
+bazel run //prow/cmd/checkconfig -- --config-path="${PROW_CONFIG}" --job-config-path="${JOBS_DIR}" --plugin-config="${PLUGIN_CONFIG}" --strict --warnings=tide-strict-branch --warnings=needs-ok-to-test --warnings=validate-owners --warnings=missing-trigger --warnings=validate-urls --warnings=unknown-fields
```

Comment:
Can we put this on more lines? Hard to read

Reply:
Yes, I'll do that in the follow up PR to enable the `mismatched-tide-lenient` check.
/hold cancel